sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/TaskUtils.java

```diff
@@ -56,6 +58,15 @@ public static PodSpec getPodSpec(ServiceSpec serviceSpec, TaskInfo taskInfo) thr
     }
 
     /**
+     * Converts the provided string to a valid environment variable name.
+     *
+     * For example: {@code hello.There!} => {@code HELLO_THERE_}
+     */
+    public static String toEnvName(String str) {
+        return ENVVAR_INVALID_CHARS.matcher(str.toUpperCase()).replaceAll("_");
```

Comment:
Lowercase is legal but unusual. I've only seen lowercase in shell scripts for internal use (ie unexported variables). Updated the javadoc to clarify.